internal/file/watch/watch_test.go

```diff
@@ -354,118 +619,107 @@ func Test_watch_Add(t *testing.T) {
 	}
 	type want struct {
-		err error
+		err  error
+		want *watch
 	}
 	type test struct {
 		name       string
 		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want, error) error
-		beforeFunc func(args)
+		checkFunc  func(want, *watch, error) error
+		beforeFunc func(*testing.T, *fields, args)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want, err error) error {
+	defaultCheckFunc := func(w want, got *watch, err error) error {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got error = %v, want %v", err, w.err)
 		}
+		if got, want := len(got.dirs), len(w.want.dirs); got != want {
+			return errors.Errorf("dirs length = %d, want %d", got, want)
+		}
+		for name := range w.want.dirs {
+			if _, ok := got.dirs[name]; !ok {
+				return errors.Errorf("dirs %s is not exists", name)
+			}
+		}
 		return nil
 	}
+	defaultBeforeFunc := func(t *testing.T, fields *fields, args args) {
+		t.Helper()
+
+		var err error
+		fields.w, err = fsnotify.NewWatcher()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           dirs: nil,
-		       },
-		       fields: fields {
-		           w: nil,
-		           eg: nil,
-		           dirs: nil,
-		           mu: sync.RWMutex{},
-		           onChange: nil,
-		           onCreate: nil,
-		           onRename: nil,
-		           onDelete: nil,
-		           onWrite: nil,
-		           onChmod: nil,
-		           onError: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           dirs: nil,
-		           },
-		           fields: fields {
-		           w: nil,
-		           eg: nil,
-		           dirs: nil,
-		           mu: sync.RWMutex{},
-		           onChange: nil,
-		           onCreate: nil,
-		           onRename: nil,
-		           onDelete: nil,
-		           onWrite: nil,
-		           onChmod: nil,
-		           onError: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "returns nil when w.w.Add returns nil",
+			args: args{
+				dirs: []string{
+					"./watch.go", "./option.go",
+				},
+			},
+			fields: fields{
+				dirs: make(map[string]struct{}),
+			},
+			want: want{
+				err: nil,
+				want: &watch{
+					dirs: map[string]struct{}{
+						"./watch.go":  struct{}{},
+						"./option.go": struct{}{},
+					},
+				},
+			},
+		},
+
+		{
+			name: "returns error when w.w.Add returns no such file or director error",
+			args: args{
+				dirs: []string{
+					"vald.go",
+				},
+			},
+			fields: fields{
+				dirs: make(map[string]struct{}),
+			},
+			want: want{
+				err: syscall.Errno(0x2),
+				want: &watch{
+					dirs: make(map[string]struct{}),
+				},
+			},
+		},
```

Comment:
Do you think we need to test if we add the same file name?
I am not sure but I think there is a potential bug that if we add the same file with different path, the same will be added twice.

for example  adding "./watch.go" and "../watch/watch.go"